plugin/contrib/mesos/pkg/executor/executor.go

```diff
@@ -231,6 +248,81 @@ func (k *KubernetesExecutor) onInitialRegistration() {
 	}
 }
 
+// InitializeStaticPodsSource blocks until initial regstration is complete and then creates a static pod source
+// using the given factory func.
+func (k *KubernetesExecutor) InitializeStaticPodsSource(sourceFactory func(string)) {
+	<-k.initialRegComplete
+	staticPodsConfigPath := k.extractStaticPodConfig()
+	if staticPodsConfigPath != "" {
+		log.V(2).Infof("initializing static pods source factory, configured at path %q", staticPodsConfigPath)
+		sourceFactory(staticPodsConfigPath)
+	}
+}
+
+// ExtractStaticPodConfig extracts a compressed archive from k.staticPodsConfig and returns
+// the path to the extracted configuration. any errors are logged and result in returning ""
+func (k *KubernetesExecutor) extractStaticPodConfig() string {
+	if k.staticPodsConfig == nil {
+		return ""
+	}
+
+	// extract to local sendBox so it is automatically cleaned up
+	dir, err := ioutil.TempDir(".", "executor-k8sm-archive")
```

Comment:
using `"."` results in pollution of the source tree when running unit tests. maybe we can reuse the kubelet's `rootDir`? and have unit tests specify a different path (`"/tmp"`?)